src/Components/3DV/SceneView.tsx

```diff
@@ -224,8 +225,53 @@ function SceneView(props: ISceneViewProps, ref) {
     debugLog('debug', 'debug', 'SceneView Render');
     const url = modelUrl === 'Globe' ? globeUrl : modelUrl;
 
-    const sortMeshesOnLoad = () => {
+    const preProcessMeshesOnLoad = () => {
```

Comment:
is it possible to pull this function out to module scope or a utility class and just pass in what you need as props?

Reply:
I'm definitely open to more discussion on this!  I think it makes sense where it is as the SceneView function configures and returns the babylon scene and this preprocess is modifying the meshes array attached to that scene. We are also setting up the originalMaterials array which is used in the various useEffects within the function. My guess is that overwriting the scene meshes array may create issues within the babylon scene as some things like TransformNode and AbstractMesh have specific functions they want you to call on the scene to add or remove them.

Reply:
@msnyder-msft Would you mind taking a look at the latest version of this?

Reply:
sure. ok. if we can't move it fine. Just trying to not make the issue worse with a 2000+ line function. 😆